_posts/2020-03-15-learning-to-share-accomplishments-2.md

```diff
@@ -0,0 +1,214 @@
+* We didn't have 1-on-1 meetings after our original manager wasn't there anymore. I tried to have unofficial 1:1s with 
+  the two original remaining team members just to make sure they were doing ok. 
+
+### Favorite Work Project: Team Metric Tracker
+
+I really liked this project not because it was particularly complex or really cool, but because of the creative 
+solutions that came out of it.
+
+Our five-person team was all hired at the same time. We were on a three month internal project which helped us learn 
+the security protocols for software at the bank. The project was sold to us as "make this Excel spreadsheet into a web 
+app". Once we dived in deep, we realized this was a quirky problem with some hidden issues. The spreadsheet had a 
+dozen tabs, data from several years, a constantly evolving report, and a dozen users that all added data to it. Our 
+basic web app turned into a quirky data project that required data auditing, a report-designing system, hierarchies of 
+management, and redefining terminology the department used. (For more fun, we were told another team tried to tackle 
+this in the past and couldn't do it.)
+
+**What did I Do?**
+
+* While many of us worked on the original design and wireframes, I came up with the idea of redefining their terminology 
+  (what we called "metrics" and different kinds of metrics) so it made sense to us (the client used a lot of words 
+  interchangeably), and later I designed a way to abstract out the database tables to make it easy to store the 
+  different kinds of metrics in the same table. It resulted in faster queries later.
+* To ensure people could edit the actual data metrics, we had a sort of "meta" database table describing the metrics 
+  themselves, then a table to store the metric data. (ex: "Widgets per hour" vs "this month we did 12 widgets/hr".) I 
+  had suggested this idea originally and it replaced an old idea that wasn't really effective.
+* Some metrics were calculated (ex: average per year might be the sum of the metric divided by 12). We wanted to ensure 
+  users couldn't mess up these formulas. I came up with two ideas. The first was to make a formula that could be stored 
+  with both constants (as in the "12" above) as well as ID numbers (so some metric might be database ID 45). Prefixing 
+  a 'c' or 'i' on the number meant we could tell what the number meant, and ID numbers could just pull the metric from 
+  it's table and row. This remained fast to query. 
+* The second idea was to provide the user a calculator-like interface to allow users to enter only valid formulas. 
+  While the calculator wasn't my idea, the validation scheme was. On the front end, we gave them a literal calculator 
+  that you could also drag metric names into the equation. This used a state machine I designed to ensure whatever 
+  they put in couldn't be saved until the formula was mathematically correct. On the back end, we used a math parsing 
```

Comment:
I love how the reader is starting to get an idea of how the project improved for the user as well as the "and it was still fast" parts. I am def getting the picture of an individual/team who is "with" their users in the experience and that gives me a lot of confidence